### PR TITLE
Allow students to bulk download materials 

### LINKED ISSUE #1270: Student cannot bulk download materials
Access Denied

### PR DESCRIPTION
**Pull request by allenwq**
This will fix #1270 

This PR also fixed an issue that files in current folder is not downloaded.

### CODE REVIEW

File: app/controllers/course/material/folders_controller.rb [hunk 1]
```diff
@@ -57,9 +57,10 @@ def upload_materials
   end
 
   def download
-    materials = @folder.descendants.map { |f| f.materials.accessible_by(current_ability) }.flatten
+    @materials = (@folder.descendants + [@folder]).
```
Comment (weiqingtoh) [inline app/controllers/course/material/folders_controller.rb]: Does it handle more multiple levels of nesting.

Comment (allenwq) [inline app/controllers/course/material/folders_controller.rb]: yes, that's what descendants does

Comment (weiqingtoh) [inline app/controllers/course/material/folders_controller.rb]: Okay, sorry didn't read the docs for this >_<

**Comment by weiqingtoh** [comment]
LGTM @kxmbrian @fonglh ?

File: app/controllers/course/material/folders_controller.rb (hunk 1 above)
Comment (fonglh) [inline app/controllers/course/material/folders_controller.rb]: How come no longer need to check the ability?

Comment (weiqingtoh) [inline app/controllers/course/material/folders_controller.rb]: The `acessible_by` is in the next line - is that what you are asking?

Comment (fonglh) [inline app/controllers/course/material/folders_controller.rb]: Yes that's right. missed it just now

**Pull request merged**